reactivesocket-core/src/main/java/io/reactivesocket/internal/Requester.java

```diff
@@ -115,8 +114,10 @@ public static Requester createServerRequester(
     }
 
     public void shutdown() {
-        // TODO do something here
-        System.err.println("**** Requester.shutdown => this should actually do something");
+        Disposable disposable = connectionSubscription.getAndSet(CANCELLED);
```

Comment:
nit: `EmptyDisposable` (`CANCELLED` here) ignores `dispose` call, so you don't need `&& disposable != CANCELLED` check here.